vmaas/tests/conftest.py

```diff
@@ -15,11 +16,45 @@ def pytest_configure(config):
     config.addinivalue_line('markers', 'smoke: mark a test as a smoke test.')
 
 
+@pytest.fixture(name='make_conn')
+def _make_conn():
+    def make_conn():
+        hostname = conf.get('hostname', 'localhost')
+        try:
+            hostname, port = hostname.split(':')
+        except ValueError:
+            port = 8080 if hostname in ('localhost', '127.0.0.1') else 80
+        return VMaaSClient(hostname, port=port)
+
+    yield make_conn
+
+
 @pytest.fixture()
-def rest_api():
+def rest_api(make_conn):
+    return make_conn()
+
+
+@pytest.fixture(scope="session", autouse=True)
```

Comment:
Why ``autouse``? This mean it will be used for **all** tests.
Also why in ``conftest.py`` if it makes sense only for tests in ``TestUpdatesDiff``. Would it work as expected when placed under ``TestUpdatesDiff`` as class scoped fixture?

Reply:
The reason is that now there is cache even for application. App is running in 2 processes as default and each process has its own cache. What I wanted to accomplish is that one process have cached response with filters and the other without filters. I need to do this at the very beginning of test suite, because during the tests there can be other tests which could cause that there will be same cached response in both processes and we won't hit the issue. Thus class scoped fixture isn't enough. 

`autouse` and `scope=session` are used because I wanted to run this at the beginning without declaring any function argument to use this fixture.

from https://docs.pytest.org/en/latest/fixture.html#scope-sharing-a-fixture-instance-across-tests-in-a-class-module-or-session
>autouse fixtures obey the scope= keyword-argument: if an autouse fixture has scope='session' it will only be run once, no matter where it is defined. 
So I expect it will run only once for session scope.